CONTRIBUTING.md

```diff
@@ -0,0 +1,182 @@
+  report.
+- Ensure that all tests pass before submitting a pull request.
+
+### Writing Tests
+
+- When adding new features or fixing bugs, write tests that cover your changes. We strive for comprehensive test
+  coverage to maintain code quality.
+- Follow our project's conventions for test structure and naming. Tests should be placed in the `tests` directory.
+- Use descriptive test function names that clearly state what is being tested.
+
+### Test Coverage
+
+- We aim for high test coverage but understand it's not always practical to achieve 100%. Focus on testing critical
+  paths and complex logic.
+- After running tests, check the coverage report to ensure your changes are adequately covered. You can generate a
+  coverage report by running `pytest --cov=src`.
+- We use [codecov](https://app.codecov.io/gh/Draegerwerk/sdc11073) to monitor test coverage. Ensure your pull request
+  does not significantly decrease the project's overall coverage.
+
+By following these guidelines, you help ensure that our project remains stable, reliable, and easy to maintain. Thank
+you for contributing to our tests!
+
+## How to create a pull request
+
+Creating a pull request is a critical step in contributing to our project. Here's how to do it effectively:
+
+1. **Ensure Your Branch is Up-to-Date**: Before starting, make sure your branch is up-to-date with the branch you're
+   planning to merge into.
+
+2. **Check Your Changes**:
+    - Run tests with `pytest` to ensure all tests pass.
+    - Use `ruff check /path/to/your/changes1 /path/to/your/changes2 ...` for static code analysis.
+    - Apply `mypy /path/to/your/changes1 /path/to/your/changes2 ...` for type checking.
+
+3. **Commit Your Changes**: Commit your changes with clear, descriptive commit messages.
+
+4. **Push to Your Fork**: Push your changes to your fork of the repository.
+
+5. **Create the Pull Request**:
+    - Navigate to the original repository you forked from.
+    - Click on the "Pull requests" tab and then the "New pull request" button.
+    - Choose your fork and the branch with your changes as the "compare" branch and the branch of our repository you
+      want to merge into as the "base" branch.
+    - Fill in the pull request form with a clear title and a detailed description of your changes.
+    - If your pull request is related to an issue or discussion thread, reference it in the description or/and under
+      the "Development" section.
+
+6. **Review and Adjust**:
+    - After submitting, at least one of our maintainers has to review your pull request. Be open to feedback also from
+      other contributors and ready to make adjustments as needed.
+    - If requested, make further commits to your branch to address feedback.
+
+7. **Acceptance and Merge**:
+    - Once your pull request is approved by a project maintainer and all CI checks pass, it will be merged.
+    - Congratulations! You've successfully contributed to the project.
+
+## Copyright
```

Comment:
Instead, point out that contributors may add their names to the CONTRIBUTORS.md file.

Reply:
done